### DISCUSSION TITLE
Wyłączenie testowania wszystkich przekaźników podczas uruchomienia

### DISCUSSION THREAD

**Discussion by lot-p** [lkankowski/arduino-multi-relay2#45] [General]
Mam pytanie jak wyłączyć testowanie wszystkich przekaźników podczas uruchomienia Andino. Uciążliwe jest podczas zaniku napięcia np. w nocy, i gdy prąd wraca to wszystkie lampy w domu się załączają i wyłączaj

**Comment by lkankowski** [comment]
Co masz na myśli pisząc "testowanie wszystkich przekaźników podczas uruchomienia Andino"?
Jako takiego testowania nie ma, natomiast dla przekaźników skonfigurowanych bez RELAY_STARTUP_ON lub RELAY_STARTUP_OFF przywracany jest poprzedni stan z pamięci EEPROM.

Ostatnio też zauważyłem, że na moim arduino używanym przez kilka lat, po restarcie zaczęły się włączać losowe przekaźniki. Mam 2 podejrzenia:
1. Pamięć EEPROM jest uszkodzona - ma ograniczoną ilość cykli zapisu i po prostu się "wypaliła"
2. W programie jest błąd - podejrzewam to już od jakiegoś czasu, ale nie jestem w stanie go wykryć. Możliwe, że buga nie ma, a mamy do czynienia z problemem z pkt 1.

Najprościej wyeliminować problem z pkt 1 to wymiana Arduino.

**Comment by goraht** [comment]
Mam dokładnie ten sam problem na dwóch różnych arduino (wykorzystuję dwa jednocześnie, do różnych świateł). Co ciekawe użycie parametru RELAY_STARTUP_OFF wydaje się być w tym wypadku zupełnie zignorowane - nie działa. Właśnie do mnie przyszła nowa płytka arduino mega r3, którą mam nadzieję przetestować w sobotę.

UPDATE
Przetestowałem za to na szybko użycie parametrów IGNORE_BUTTONS_START_MS=2000 oraz EEPROM_OFFSET=100 i nagle stan między zanikami prądu zaczął być zapamiętywany oraz RELAY_STARTUP_OFF też zaczął działać. Dzięki za tą podpowiedź w dokumentacji. Niestety "testowanie przekaźników" nadal ma miejsce. Na bardzo krótko, zaraz po powrocie zasilania, wszystkie przekaźniki zapalają się na ułamek sekundy.

**Reply by lkankowski** [reply]
Ten ułamek sekundy to może być stan początkowy Arduino i nie mamy na to wpływu - po podłączeniu zasilania piny są zawsze w jakimś stanie (zależy od modelu) i zanim program wystartuje, a następnie ustawi odpowiedni stan wg konfiguracji, a potem odczyta z EEPROM stan poprzedni mija trochę czasu.
Inna kwestia to błąd w programie, czego już wcześniej nie wykluczałem - najprostsze sprawdzenie to instalacja poprzedniej wersji: https://github.com/lkankowski/arduino-multi-relay2/releases/tag/v2.3.2 (przy założeniu, że i tam nie było błędu 😄)
